Sources/Configuration/DefaultConfigurationManager.swift

```diff
@@ -0,0 +1,160 @@
+        public static let refreshCheckIntervalSeconds = 60.0 // check if we need a refresh every minute
+
+        static let lastUpdateDefaultsKey = "configuration.lastUpdateTime"
+    }
+
+    private var defaults: KeyValueStoring
+
+    public var fetcher: ConfigurationFetching
+    public var store: ConfigurationStoring
+
+    public init(fetcher: ConfigurationFetching, store: ConfigurationStoring, defaults: KeyValueStoring = UserDefaults()) {
+        self.fetcher = fetcher
+        self.store = store
+        self.defaults = defaults
+        super.init()
+        NSFileCoordinator.addFilePresenter(self)
+    }
+
+    deinit {
+        NSFileCoordinator.removeFilePresenter(self)
+    }
+
+    public static let queue: DispatchQueue = DispatchQueue(label: "Configuration Manager")
+    public static let filePresenterOperationQueue = OperationQueue()
+
+    public var lastUpdateTime: Date {
+        get {
+            defaults.object(forKey: Constants.lastUpdateDefaultsKey) as? Date ?? .distantPast
+        }
+        set {
+            defaults.set(newValue, forKey: Constants.lastUpdateDefaultsKey)
+        }
+    }
+
+    private var timerCancellable: AnyCancellable?
+    public var lastRefreshCheckTime: Date = Date()
+
+    public func start() {
+        Logger.config.debug("Starting configuration refresh timer")
+        timerCancellable = Timer.publish(every: Constants.refreshCheckIntervalSeconds, on: .main, in: .default)
+            .autoconnect()
+            .receive(on: Self.queue)
+            .sink(receiveValue: { _ in
+                self.lastRefreshCheckTime = Date()
+                self.refreshIfNeeded()
+            })
+        Task {
+            await refreshNow()
+        }
+    }
+
+    /// Implement this in the subclass.
+    /// Use this method to fetch neccessary configurations and store them.
+    open func refreshNow(isDebug: Bool = false) async {
+        fatalError("refreshNow Must be implemented by subclass")
+    }
+
+    @discardableResult
+    private func refreshIfNeeded() -> Task<Void, Never>? {
+        guard isReadyToRefresh else {
+            Logger.config.debug("Configuration refresh is not needed at this time")
+            return nil
+        }
+        return Task {
+            await refreshNow()
+        }
+    }
+
+    open var isReadyToRefresh: Bool { Date().timeIntervalSince(lastUpdateTime) > Constants.refreshPeriodSeconds }
+
+    public func forceRefresh(isDebug: Bool = false) {
+        Task {
+            await refreshNow(isDebug: isDebug)
+        }
+    }
+
+    /// Will try to update the config again at the regularly scheduled interval
+    /// **Note:** You must call `start()` on your `ConfigurationManager` instance for this to take effect. It relies on the internal refresh loop of the
+    /// `DefaultConfigurationManager` class
+    public func tryAgainLater() {
+        lastUpdateTime = Date()
```

Comment:
I'll need to check how we use this, right now it's not thread safe but I understand that it should always be called from a specific queue - maybe we can use a dispatch assertion. I'll take a look at the clients and report back.